pkg/aws/aws.go

```diff
@@ -88,6 +93,37 @@ func DefaultProfile() string {
 	return "default"
 }
 
+func (ac *awsCloud) setSuffixes(vpcID string) error {
+	if ac.nodeSGSuffix != "" {
+		return nil
+	}
+
+	publicSubnets, err := ac.findPublicSubnets(vpcID, ac.filterByName("{infraID}*-public-{region}*"))
+	if err != nil || len(publicSubnets) == 0 {
```

Comment:
Do we want to return nil if err == nil && len(publicSubnets) == 0 ?

Reply:
We should be finding a public subnet else thee rest of the logic may not work.